datasets/glossis/interpolate_vtk.py

```diff
@@ -0,0 +1,248 @@
+
+    configure_input(resample, cell2point)
+
+    resample.sampling_dimensions = np.array([width, height, 1])
+    resample.sampling_bounds = np.array([-180, 180, -90, 90, 0, 0.0])
+    resample.use_input_bounds = False
+    # TODO: check if this can be left out
+    resample.modified()
+    resample.update()
+    polydata.modified()
+
+    return resample
+
+def resample_data(netcdfs, img_size, t, variables):
+    """ Resample data for timestep, variables in netcdf to given image size """
+    polydatas_by_name = {}
+
+    mbd = tvtk.MultiBlockDataSet()
+    mbd.number_of_blocks = len(netcdfs)
+
+    masks_by_name = {}
+
+    for i, f in enumerate(netcdfs):
+        ds = netCDF4.Dataset(f)
+        grid = gridded.pyugrid.ugrid.UGrid.from_nc_dataset(ds)
+        polydata, mask =  grid2poly(grid)
+        mbd.set_block(i, polydata)
+        masks_by_name[f.name] = mask
+        polydatas_by_name[f.name] = polydata
+        ds.close()
+
+    mbm = tvtk.MultiBlockMergeFilter()
+    configure_input_data(mbm, mbd)
+    resample = setup_pipeline(mbm, width=img_size[0], height=img_size[1])
+
+    for path in netcdfs:
+        polydata = polydatas_by_name[path.name]
+        mask = masks_by_name[path.name]
+        update_data(path, polydata, mask, variables, t=0)
+    resample.update()
+    mbm.update()
+
+    return resample
+
+def fm_to_tiff_vtk(
+    bucketname,
+    prefixname,
+    tmpdir,
+    variables=["currents_u", "currents_v"],
+    filter="currents",
+    output_fn="glossis_currents",
+    nodata=-9999,
+    extra_bands=0,
+):
+    """Convert FM netcdfs in bucket into geotiffs for each timestep using vtk."""
+
+    # Download netcdfs files from bucket
+    netcdfs = download_netcdfs_from_bucket(
+        bucketname, prefixname, tmpdir, filter)
+    netcdfs = [pathlib.Path(netcdf) for netcdf in netcdfs]
+
+    path = pathlib.Path('tmp/netcdfs')
+    netcdfs = list(path.glob('*.nc'))
+    
+    if len(netcdfs) == 0:
+        logger.warning(f"No {filter} files found!")
+        return []
+
+    # Determine timesteps in first netcdf
+    nc = netCDF4.Dataset(netcdfs[0])
+    timesteps = netCDF4.num2date(
+        nc.variables["time"][:], units=nc.variables["time"].units
+    )
+    logger.info(
+        "{} timesteps of which only the first six will be processed.".format(
+            len(timesteps)
+        )
+    )
+
+    metadata = nc.__dict__
+    date_created = datetime.strptime(
+        metadata["date_created"], "%Y-%m-%d %H:%M:%S %Z"
+    )
+    metadata["date_created"] = datetime.strftime(
+        date_created, "%Y-%m-%dT%H:%M:%S")
+    analysis_time = netCDF4.num2date(
+        nc.variables["analysis_time"][:], units=nc.variables["analysis_time"].units
+    )[0]
+
+    img_size = (7200, 7200)
+
+    # After 6 timesteps (of an hour) we reach the timestamp
+    # that will be overwritten by the next 6hourly run.
+    timesteps = timesteps[0:6]
+    tiff_files = []
+
+    for ti, time in enumerate(timesteps):
```

Comment:
Each timestep requires opening all the files, setting up the grid and workflow etc, while only the values on the grid change. Would at least add a TODO about it.

Reply:
Yes I know, actually would like to fix this but will at least put a comment in the code. But it only takes a couple of seconds to setup the grid so it is not that relevant to fix this.